internal/charmstore/store.go

```diff
@@ -788,3 +797,44 @@ func (s *Store) SynchroniseElasticsearch() error {
 	}
 	return nil
 }
+
+//GetPromulgatedURL finds the promulgatedURL that should be used for
+//this newly uploaded charm, if the charm should be promulgated,
+//othewise it returns nil. An error is returned if there is a problem
+//communicating with the storage.
+func (s *Store) GetPromulgatedURL(id *charm.Reference) (*charm.Reference, error) {
```

Comment:
I'm not entirely sure that this is the best place for this method. It seems like something that's probably only single use (by the archive POST code), and the current logic for incrementing the revision lives in v4/archive.go. What do you think about moving it there?